src/Arcus.WebApi.Security.Authentication/SharedAccessKey/SharedAccessKeyAuthenticationFilter.cs

```diff
@@ -53,35 +55,55 @@ public async Task OnAuthorizationAsync(AuthorizationFilterContext context)
             Guard.For<ArgumentException>(() => context.HttpContext.Request.Headers == null, "Invalid action context given without any HTTP request headers");
             Guard.For<ArgumentException>(() => context.HttpContext.RequestServices == null, "Invalid action context given without any HTTP request services");
 
-            if (context.HttpContext.Request.Headers
-                       .TryGetValue(_headerName, out StringValues requestSecretHeaders))
+            string foundSecret = await GetAuthorizationSecret(context);
+
+            if (!context.HttpContext.Request.Headers.ContainsKey(_headerName) && !context.HttpContext.Request.Query.ContainsKey(_queryParameterName))
+            {
+                context.Result = new UnauthorizedResult();
+            }
+            else
             {
-                ISecretProvider userDefinedSecretProvider = 
-                    context.HttpContext.RequestServices.GetService<ICachedSecretProvider>()
-                    ?? context.HttpContext.RequestServices.GetService<ISecretProvider>();
-                
-                if (userDefinedSecretProvider == null)
+                if (!String.IsNullOrWhiteSpace(_headerName) && context.HttpContext.Request.Headers
```

Comment:
Can we add a test that ensure that different casing for the parameter still works?

Reply:
Added